test/unit/math/prim/fun/ones_vector_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathFunctions, ones_vector) {
+  using Eigen::VectorXd;
+  using stan::math::ones_vector;
+
+  VectorXd u0 = ones_vector(0);
```

Comment:
[optional]
This can also all go in a loop:

```
for (int K = 0; K < 5; ++K) {
  Eigen::VectorXd y = Eigen::Constant(1, K);  // or whatever the right syntax for this is
  expect_float_eq(y, ones_vector(K));
}
```

It doesn't matter if that's how you implemented the `ones_vector` function as it's clearly the right thing to test.  The point is to have the test be clear and right.